File: app/src/main/java/com/example/weather/WeatherAPI.java
package com.example.weather;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface WeatherAPI {
    @GET("/data/2.5/onecall?")
    public Call<Info> GetSityInf(@Query(value = "lat", encoded = false) Double lat,
                                 @Query(value = "lon", encoded = false) Double lon,
                                 @Query(value = "lang", encoded = false) String lang,
                                 @Query(value = "appid", encoded = false) String KEY,
                                 @Query(value = "units", encoded = false) String units);



//    public Call<Info>GetSityInf(@Path("?lat") Double lat, @Path("lon") Double lon);

}

File: lib/data/api/weather/weather_api.dart
import 'dart:convert';
import 'dart:ffi';

import 'package:dio/dio.dart';
import 'package:weather/core/constants/constant.dart';

class WeatherApi {
  static Future<Response?> getWeather(String token, double lat, double lon) async {
    var dio = Dio();
    Response? news;
    try {
      news = await dio.get('${AppConstants.base}/data/2.5/forecast',
          queryParameters: {"appid": token ,'lat': lat, 'lon': lon, 'units' : 'metric', 'lang': 'ru'});

      print('STATUS ${news.statusMessage}');
      print('DATA ${news.data}');
      return news;
    } catch (e) {
      print(e);
      rethrow;
    }
  }
}
